src/main/java/uk/ac/cam/cl/dtg/segue/api/UsersFacade.java

```diff
@@ -709,30 +708,23 @@ private Response updateUserObject(final HttpServletRequest request, final HttpSe
                 }
             }
 
-            // Now update the email preferences
-            emailPreferenceManager.saveEmailPreferences(userObjectFromClient.getId(), emailPreferences);
+            if (userPreferences != null) {
 
-            //----------------------------------------------------------------------------------------------------------
-            // FIXME - the code between the dashed lines should be refactored out of this class; it does not belong here!
```

Comment:
My solution to this would be to have user preferences with their allowed values to be held in a config file somewhere, rather than hardcoding them in. I would still need to implement this if we all agree its the best solution.